### PR TITLE
Implement GitHub Actions on PR

### PR DESCRIPTION
**Pull request by aaronhaefner**
This PR adds two files to `.github/workflows` to run as tests triggered upon pull request or push to main.

### CODE REVIEW

**Comment by aaronhaefner** [comment]
@michplunkett any idea how to fix the issue below? The actions are working except for this bug I'm seeing causing the checks to fail. I purposely was failing the test action but expected it to work after making the change in 7278f99680f7d5672cf8cb472e84ed54d56fb799.

```
MacBook-Pro-3:new-arrivals-chi aaronhaefner$ poetry install --no-interaction
Installing dependencies from lock file

No dependencies to install or update

/Users/aaronhaefner/chicago_projects/new-arrivals-chi/new_arrivals_chi does not contain any element
```

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# IDE files and misc.
+.idea
+.vscode
+**/.DS_Store
```
Comment (michplunkett) [inline .gitignore:4]: Added these to save you some time when checking files in.

File: pyproject.toml [hunk 2]
```diff
@@ -4,7 +4,6 @@ version = "0.1.0"
 description = ""
 authors = ["Aaron Haefner <aaronhaefner@uchicago.edu>"]
 readme = "README.md"
-packages = [{include = "new_arrivals_chi"}]
```
Comment (michplunkett) [inline pyproject.toml:7]: The problem stemmed from two things:
1. You had this section here, which I don't believe you need. You are not gonna export this as a package at some point, as it's an application.
2. You lacked a base python package in your repository, so `pytest` didn't know what to reference as your non-test files. That was corrected by creating the `new_arrivals_chi` package that you can use as the base folder for your application.

File: .gitignore [hunk 3]
```diff
@@ -153,10 +158,3 @@ dmypy.json
 
 # Cython debug symbols
 cython_debug/
-
-# PyCharm
-#  JetBrains specific template is maintained in a separate JetBrains.gitignore that can
-#  be found at https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
-#  and can be added to the global gitignore or merged into this file.  For a more nuclear
-#  option (not recommended) you can uncomment the following to ignore the entire idea folder.
-#.idea/
```
Comment (michplunkett) [inline .gitignore:162]: This was redundant with the addition of the above ignore statements.

**Review by klinkoberstar — APPROVED** [review]

**Comment by aaronhaefner** [comment]
Michael was the latest commit, so I merged this in on my and Kathryn's approval.

**Pull request merged**